File: backend/common/models/project_deployment_log.js
const mongoose = require('mongoose');

const projectDeploymentLogSchema = mongoose.Schema({
    projectId: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'Project',
        required: true
    },
    logs: {
        type: Buffer,
        required: true
    }
}, {
    timestamps: true
});

projectDeploymentLogSchema.set('toJSON', {
  transform: function (doc, ret) {
    if (ret.createdAt) {
      ret.createdAt = ret.createdAt.toISOString();
    }
    if (ret.updatedAt) {
      ret.updatedAt = ret.updatedAt.toISOString();
    }
    return ret;
  }
});

const ProjectDeploymentLog = mongoose.model('ProjectDeploymentLog', projectDeploymentLogSchema);
module.exports = ProjectDeploymentLog;

File: frontend/lib/features/project_deployment_logs/domain/entities/project_deployment_log.dart
class ProjectDeploymentLog {
  final String log;
  final DateTime createdAt;

  ProjectDeploymentLog({
    required this.log,
    required this.createdAt,
  });

  @override
  String toString() {
    return 'ProjectDeploymentLog{log: $log, createdAt: $createdAt}';
  }
}
